libgalois/src/analytics/pagerank/pagerank-pull.cpp

```diff
@@ -315,7 +303,7 @@ PagerankPullResidual(
   residual.allocateInterleaved(pg->num_nodes());
 
   InitNodeDataResidual(&graph, delta, residual, plan);
-  ComputeOutDeg(&graph);
+  //ComputeOutDeg(&graph); TODO(lhc) just test
```

Comment:
Comment is confusing. Sorry about that. This is because residual is not using separate node data. The current `ComputeOutDeg()` sets `node_data`, not `PropertyGraph`. Interface should be differentiated. If you fine with my plan, I will do.

Reply:
Add a more detailed comment. TODOs in code should be self-explanatory.

Reply:
I just added overloading and removed that comment. That looks safe.